test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env bash
+##############################################################################################
+# Check branches to ensure backward compatibility.
```

Comment:
With this PR this script covers not only branch checking (also wiredtiger release checking) and backward compatibility (also forward compatibility and upgrade/downgrade).

Reply:
Description updated in 758c906d6.